server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_delete_nonexisting_importer(self, mock_factory):
+        """
+        Attpempt diassociate an importer that is not associated to a repository.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_importer = {"id": "mock_importer"}
+        mock_factory.repo_importer_manager.return_value.get_importer.return_value = mock_importer
+
+        repo_importer = RepoImporterResourceView()
+        try:
+            repo_importer.delete(mock_request, 'mock_repo', 'not_mock_importer')
+        except pulp_exceptions.MissingResource, response:
+            pass
+        else:
+            raise AssertionError("MissingResource should be raised if importer is not associated "
+                                 "with the repo.")
+
+        self.assertEqual(response.http_status_code, 404)
+        self.assertTrue(response.error_code is error_codes.PLP0009)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_UPDATE())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.repo_importer_manager.update_importer_config')
+    @mock.patch('pulp.server.webservices.views.repositories.tags')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_put_update_importer(self, mock_factory, mock_tags, mock_update_importer, mock_resp):
+        """
+        Update an importer with all required params.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps({'importer_config': 'test'})
+        mock_importer = {"id": "mock_importer"}
+        mock_factory.repo_importer_manager.return_value.get_importer.return_value = mock_importer
+        mock_task = [mock_tags.resource_tag(), mock_tags.resource_tag(), mock_tags.action_tag()]
+
+        repo_importer = RepoImporterResourceView()
+        try:
+            repo_importer.put(mock_request, 'mock_repo', 'mock_importer')
+        except pulp_exceptions.OperationPostponed, response:
+            pass
+        else:
+            raise AssertionError("OperationPostponed should be raised for update importer task")
+
+        self.assertEqual(response.http_status_code, 202)
+        mock_update_importer.apply_async_with_reservation.assert_called_once_with(
+            mock_tags.RESOURCE_REPOSITORY_TYPE, 'mock_repo', ['mock_repo'],
+            {'importer_config': 'test'}, tags=mock_task
+        )
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_UPDATE())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
```

Comment:
remove unnecessary mock like generate_json_response_with_pulp_encoder, tags and update_importer_config